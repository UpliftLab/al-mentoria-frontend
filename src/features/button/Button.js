import PropTypes from 'prop-types';

const Button = ({
  child,
  onClick,
  isSubmit,
  isWhite,
}) => {
  const classesForGreen = 'bg-lime-500 text-white hover:bg-lime-400';
  const classesForWhite = 'bg-white text-lime-500 hover:bg-transparent hover:text-white hover:border-white';

  return (
    <button
      type={isSubmit ? 'submit' : 'button'}
      onClick={onClick}
      className={`${isWhite ? classesForWhite : classesForGreen} px-6 py-2 rounded-full font-semibold min-w-[10rem] transition-colors border-2 border-transparent`}
    >
      {child}
    </button>
  );
};

Button.propTypes = {
  child: PropTypes.node.isRequired,
  onClick: PropTypes.func,
  isSubmit: PropTypes.bool,
  isWhite: PropTypes.bool,
};

Button.defaultProps = {
  onClick: () => { },
  isSubmit: false,
  isWhite: false,
};

export default Button;
