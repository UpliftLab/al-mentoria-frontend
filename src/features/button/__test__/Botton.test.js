import {
  render,
  screen,
  cleanup,
  fireEvent,
} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Button from '../Button';

describe('Button', () => {
  afterEach(cleanup);

  it('renders correctly in the DOM', () => {
    const tree = renderer
      .create(<Button child="click" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with the green background', () => {
    render(
      <Button child="click" />,
    );

    const button = screen.getByRole('button');
    const { classList } = button;

    expect(classList).toContain('bg-lime-500');
  });

  it('renders with the white background', () => {
    render(
      <Button child="click" isWhite />,
    );

    const button = screen.getByRole('button');
    const { classList } = button;

    expect(classList).toContain('bg-white');
  });

  it('renders with the correct text', () => {
    render(<Button child="click" />);

    const button = screen.getByRole('button');

    expect(button.textContent).toContain('click');
  });

  it('renders with the button type by default', () => {
    render(<Button child="click" />);

    const button = screen.getByRole('button');

    expect(button.type).toBe('button');
  });

  it('renders with the submit type when isSubmit is true', () => {
    render(<Button child="click" isSubmit />);

    const button = screen.getByRole('button');

    expect(button.type).toBe('submit');
  });

  it('renders with the submit type when isSubmit is true', () => {
    render(<Button child="click" isSubmit />);

    const button = screen.getByRole('button');

    expect(button.type).toBe('submit');
  });

  it('fires the callback function when clicked', () => {
    render(<Button child="click" onClick={() => { render(<p data-testid="text">You clicked</p>); }} isSubmit />);

    const button = screen.getByRole('button');
    fireEvent.click(button);

    const text = screen.getByTestId('text');

    expect(text).toHaveTextContent('You clicked');
  });
});
