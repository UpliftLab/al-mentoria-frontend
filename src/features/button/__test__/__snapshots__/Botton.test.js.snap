// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly in the DOM 1`] = `
<button
  className="bg-lime-500 text-white hover:bg-lime-400 px-6 py-2 rounded-full font-semibold min-w-[10rem] transition-colors border-2 border-transparent"
  onClick={[Function]}
  type="button"
>
  click
</button>
`;
